The problem emerges when running `pytest tests/meltano/core tests/meltano/core/tracking`:

```
FAILED tests/meltano/core/plugin/singer/test_tap.py::TestSingerTap::test_run_discovery - AttributeError: 'BoundLoggerFilteringAtNotset' object has no attribute 'isEnable...
FAILED tests/meltano/core/plugin/singer/test_tap.py::TestSingerTap::test_run_discovery_failure - AttributeError: 'BoundLoggerFilteringAtNotset' object has no attribute '...
FAILED tests/meltano/core/plugin/singer/test_tap.py::TestSingerTap::test_run_discovery_stderr_output - AttributeError: <BoundLoggerLazyProxy(logger=None, wrapper_class=N...
FAILED tests/meltano/core/plugin/singer/test_tap.py::TestSingerTap::test_run_discovery_handle_io_exceptions - AssertionError: Regex pattern 'mock readline exception' doe...
FAILED tests/meltano/core/plugin/singer/test_tap.py::TestSingerTap::test_run_discovery_utf8_output - AttributeError: <BoundLoggerLazyProxy(logger=None, wrapper_class=Non...
FAILED tests/meltano/core/tracking/test_tracking.py::test_get_snowplow_tracker_invalid_endpoint - assert 0 == 2
```

When running `tests/meltano/core/logging/test_output_logger.py tests/meltano/core/plugin/singer/test_tap.py tests/meltano/core/tracking/`, we only see the first 5 failures - the one from `test_get_snowplow_tracker_invalid_endpoint` goes away.

--- a/tests/meltano/core/logging/test_output_logger.py
+++ b/tests/meltano/core/logging/test_output_logger.py
@@ -32,6 +32,7 @@ class TestOutputLogger:
 
     @pytest.fixture(autouse=True)
     def fixture_configure_structlog(self, log_output):
+        original_config = structlog.get_config()
         structlog.configure(
             processors=[log_output],
             logger_factory=structlog.stdlib.LoggerFactory(),
@@ -40,7 +41,7 @@ class TestOutputLogger:
         try:
             yield
         finally:
-            structlog.reset_defaults()
+            structlog.configure(**original_config)
 
     @pytest.fixture(name="redirect_handler")
     def redirect_handler(self, subject: OutputLogger) -> logging.Handler:

